internal/provider/setup_key_resource.go

```diff
@@ -0,0 +1,348 @@
+	resp.TypeName = req.ProviderTypeName + "_setup_key"
+}
+
+func (r *SetupKey) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Create and Manage Setup Keys, see [NetBird Docs](https://docs.netbird.io/how-to/register-machines-using-setup-keys) for more information.",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "SetupKey ID",
+				PlanModifiers:       []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+			},
+			"name": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "SetupKey Name",
+				PlanModifiers:       []planmodifier.String{stringplanmodifier.RequiresReplace()},
+			},
+			"expires": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "SetupKey Expiration Date",
+				PlanModifiers:       []planmodifier.String{stringplanmodifier.RequiresReplace()},
+			},
+			"expiry_seconds": schema.Int32Attribute{
+				Required:            true,
+				MarkdownDescription: "Expiry time in seconds",
+				PlanModifiers:       []planmodifier.Int32{int32planmodifier.RequiresReplace()},
+			},
+			"updated_at": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "Creation timestamp",
+				PlanModifiers:       []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+			},
+			"last_used": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "Last usage time",
+				PlanModifiers:       []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+			},
+			"key": schema.StringAttribute{
+				Computed:            true,
+				Sensitive:           true,
+				MarkdownDescription: "Plaintext setup key",
+				PlanModifiers:       []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+			},
+			"type": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "",
+				PlanModifiers:       []planmodifier.String{stringplanmodifier.RequiresReplace()},
+			},
+			"usage_limit": schema.Int32Attribute{
+				Computed:            true,
+				Optional:            true,
+				MarkdownDescription: "",
+				PlanModifiers:       []planmodifier.Int32{int32planmodifier.UseStateForUnknown()},
+				Default:             int32default.StaticInt32(0),
+			},
+			"used_times": schema.Int32Attribute{
+				Computed:            true,
+				MarkdownDescription: "",
+				PlanModifiers:       []planmodifier.Int32{int32planmodifier.UseStateForUnknown()},
+			},
+			"state": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "",
+				PlanModifiers:       []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+			},
+			"auto_groups": schema.ListAttribute{
```

Comment:
This is marked as `read-only` in the documentation, but can actually be modified in the Dashboard/API.

Reply:
Resolving in https://github.com/netbirdio/terraform-provider-netbird/pull/7